### PR TITLE
[Batch mode] Write out filemaps for supplementary outputs.

### PR DESCRIPTION
**Pull request by davidungar**
Add -driver-force-one-batch-repartition and enhance  batch_mode-overlong_argv test.

Communicate supplementary outputs from driver to frontend via file map. This PR fixes a Unicode character issue that caused the last incarnation to fail a test.

rdar://38157859

### CODE REVIEW

**Comment by davidungar** [comment]
@swift-ci please test

**Comment by davidungar** [comment]
@swift-ci Please Test Source Compatibility

**Comment by davidungar** [comment]
@graydon @jrose-apple Trying again with the right escape function! Thanks.

**Comment by jrose-apple** [comment]
I do suggest adding a Unicode filename test case this time. Maybe even including an outside-the-BMP emoji.

**Comment by davidungar** [comment]
@swift-ci please test os x platform

**Comment by davidungar** [comment]
@swift-ci please smoke test linux platform

**Comment by davidungar** [comment]
@swift-ci please test os x platform

**Review by graydon — APPROVED** [review]

File: test/Driver/batch_mode_with_supplementary_filelist_unicode.swift [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// RUN: %empty-directory(%t)
+// RUN: echo 'print("Hello, World!")' >%t/main.swift
+// RUN: touch %t/𝔼-file-01.swift %t/😂-file-02.swift %t/Ω-file-03.swift
+//
+// If the supplementary output file map does not escape the characters in the
+// source files, the frontend won't recognize the desired outputs.
+//
+// RUN: (cd %t && %target-build-swift -c -save-temps -emit-dependencies -emit-loaded-module-trace -emit-loaded-module-trace-path %t/lmt -serialize-diagnostics -driver-use-filelists -j2 main.swift  𝔼-file-01.swift 😂-file-02.swift Ω-file-03.swift  -module-name mod)
```
Comment (jrose-apple) [inline test/Driver/batch_mode_with_supplementary_filelist_unicode.swift]: What's the point of using `-emit-loaded-module-trace-path` if you don't test the output? And doesn't that only work in single-compile mode anyway?

Comment (jrose-apple) [reply test/Driver/batch_mode_with_supplementary_filelist_unicode.swift]: This shouldn't need `-save-temps` either; there are no temporaries.

**Comment by graydon** [comment]
(Followup on the watchos failure here: the failure was due to a harmless warning showing up in a shouldBeEmpty file, which in turn led us into noticing that there was a failure to propagage YAML-initiated errors back up to the OFM-parser level. Pushed two changes here that correct that + fix the comments @jrose-apple made above)

**Comment by graydon** [comment]
@swift-ci please test

**Review by jrose-apple — APPROVED** [review]

**Pull request merged**